### PR TITLE
[Add]:Header component

### LINKED ISSUE #27: Header component
## Feature Description
The Header allows users to navigate between the sections. Upon selecting one of the major sections (e.g. Datasets), the Header transitions to the variant of the section. The back button in the header takes the user back to the main dashboard.

Here are the sections:

- Datasets: `Datasets`, `Requests`, `Tensors`
- Models
- Users: `Users`, `Groups`, `Roles`
- Infrastructure: `Infrastructure`, `Nodes`, `Workers`
- Finances
- Settings: `Settings`, `Infrastructure`, `Associations`, `Monetization`, `Branding`

Models and Finances have no subsection and, consequently, no transition.

A started for the Header exists in [src/components/header/index.tsx](https://github.com/OpenMined/pygrid-admin/blob/dev/src/components/header/index.tsx) and that should be used as the starting point for this issue. The PyGrid logo and Header height have been implemented.

This is how it should look like: [Framer link](https://framer.com/share/PyGrid-Admin--sqLTLbsgzBzUix5DmV7g#r7YHL2hBg).

### PR DESCRIPTION
**Pull request by infern018**
## Description
Closes #27 
Added the header(navbar) component for each route in the index.txt file with proper functionality and routing of each section/subsection, also currently I was not able to add the adequate CSS as there were some problems in linking the global CSS file, so for mobile view CSS is not accurate, for desktop it is set correctly...

## Screenshots
Screenshot (374)
Screenshot (375)
Screenshot (376)

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by infern018** [comment]
@tcp  pls review!

**Comment by infern018** [comment]
I've resolved the conflicts now, can you have a look and tell if me there should be any changes @tcp ?

**Comment by infern018** [comment]
I have fixed some UI components and made the header responsive.

## Screenshots:
ad1
ad2

File: src/components/header/index.tsx [hunk 1]
```diff
@@ -1,15 +1,110 @@
...
   return (
+
+    
+
     <nav className="sticky top-0">
       <div className="flex flex-row flex-wrap items-center justify-between px-3 shadow bg-gray-50">
-        <div className="container mx-auto">
-          <NextLink href="/" passHref>
-            <Image className="cursor-pointer" src="/assets/logo.png" alt="PyGrid home" width={50} height={50} />
+        <div className="container w-full md:w-auto md:flex flex-row">
```
Comment (tcp) [inline src/components/header/index.tsx:87]: The header width follows the rest of the app. Changing to `w-full` and `w-auto` affects this behaviour.

File: src/components/header/index.tsx [hunk 2]
```diff
@@ -1,15 +1,110 @@
...
+
+    else
+    {
+      routes = [];
+    }
+
+    routes.forEach((route) => {
+      if(curr_path === 'dashboard')
+      {
+        var route_name = route.charAt(0).toUpperCase() + route.slice(1);
```
Comment (tcp) [inline src/components/header/index.tsx:58]: Use the `text-transform: capitalize` property instead. In tailwind, the class is `capitalize`.

**Review by tcp — CHANGES_REQUESTED** [review]
Thank you for your submission!

I've left a few comments. Please don't forget to lint and format your code following the project settings.

Feel free to push a revised PR. Take care!

File: src/components/header/index.tsx [hunk 3]
```diff
@@ -1,15 +1,110 @@
...
+      else
+      {
+        var route_name = route.charAt(0).toUpperCase() + route.slice(1)
+        list.push(<NextLink href={curr_route+'/'+route} passHref><span className="cursor-pointer block md:inline-block text-black-900 hover:text-black-600 px-3 py-3 border-b-2 border-black-900 md:border-none">{route_name}</span></NextLink>);
+      }
+      
+    })
+
+    if(process.browser){
+        document.getElementById("hamburger").onclick = function toggleMenu() {
```
[the remark this answers is not in the document]
Comment (infern018) [reply src/components/header/index.tsx:70]: Hey Thiago, had a doubt here, I will need to manipulate the DOM as I want to add that hamburger menu functionality which involves the toggle of navbar in the mobile-view,

[the remark this answers is not in the document]
Comment (tcp) [reply src/components/header/index.tsx:70]: Hey @infern018! You should use the attributes, but like this:
```js
  <Hamburger onClick={open} />
```

Check out some info on React and the (virtual) DOM: [here](https://reactjs.org/docs/faq-internals.html), [here](https://reactjs.org/docs/rendering-elements.html) and [here](https://reactjs.org/docs/react-dom.html).

[the remark this answers is not in the document]
Comment (infern018) [reply src/components/header/index.tsx:70]: I can make the `onclick` function work without using DOM, but for :
`const navToggle = **document.getElementsByClassName("toggle")**;`
          `for (let i = 0; i < navToggle.length; i++) {`
            `navToggle.item(i).classList.toggle("hidden");`
          `}`
  I will need to get access to the `toggle` class

[the remark this answers is not in the document]
Comment (tcp) [reply src/components/header/index.tsx:70]: One option is to use a state variable to keep track of this information.

```js
function Header() {
  ...
  const [isHidden, hide] = React.useState(true)
  const menuRef = React.useRef()
  ...
  <Hamburger ref={ref} onClick={() => hide(false)} />
  ...
}
```

You can use a hook for listening to clicks outside of the icon that would trigger the `() => hide(true)` to close the menu, for example.

File: src/components/header/index.tsx [hunk 4]
```diff
@@ -1,15 +1,110 @@
...
+            <Image className="cursor-pointer block md:inline-block px-3 py-3 mt-5 border-b-2 md:border-none" src="/assets/logo.png" alt="PyGrid home" width={50} height={50}/>
           </NextLink>
+
+          {/* <NextLink href="/" passHref >
+              <FontAwesomeIcon className="cursor-pointer" icon={faCaretLeft} color="gray.400"/>
+          </NextLink> */}
+
+          <div className="flex md:hidden float-right pt-3">
+          <button id="hamburger">
+            <img className="toggle block" src="https://img.icons8.com/fluent-systems-regular/2x/menu-squared-2.png" width={30} height={30} />
```
[the remark this answers is not in the document]
Comment (infern018) [reply src/components/header/index.tsx:98]: image

Will SVG if this do?

**Pull request closed without merging**